modules/custom/az_cas/az_cas.module

```diff
@@ -97,6 +107,84 @@ function az_cas_form_user_login_form_alter(&$form, FormStateInterface $form_stat
   }
 }
 
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * Modify the CAS settings form to prevent disabling auto_register when
+ * AZ CAS guest mode is enabled.
+ */
+function az_cas_form_cas_settings_alter(&$form, FormStateInterface $form_state) {
+  $az_cas_config = \Drupal::config('az_cas.settings');
+  $guest_mode = $az_cas_config->get('guest_mode');
+
+  if ($guest_mode) {
+    // Add a notice about guest mode requiring auto_register.
+    $form['user_accounts']['auto_register']['#description'] .= '<br><strong>' . t('This setting is required for AZ CAS guest authentication mode and cannot be disabled while guest mode is enabled.') . '</strong>';
+
+    // Force auto_register to be enabled and disable the checkbox.
+    $form['user_accounts']['auto_register']['#default_value'] = TRUE;
+    $form['user_accounts']['auto_register']['#disabled'] = TRUE;
+
+    // Add a custom validation to ensure auto_register remains enabled.
+    $form['#validate'][] = 'az_cas_cas_settings_validate';
+  }
+}
+
+/**
+ * Custom validation for CAS settings form.
+ *
+ * Ensures auto_register remains enabled when guest mode is active.
+ */
+function az_cas_cas_settings_validate($form, FormStateInterface $form_state) {
+  $az_cas_config = \Drupal::config('az_cas.settings');
+  $guest_mode = $az_cas_config->get('guest_mode');
+
+  if ($guest_mode && !$form_state->getValue(['user_accounts', 'auto_register'])) {
+    $form_state->setErrorByName('user_accounts][auto_register', t('Auto register users cannot be disabled while AZ CAS guest authentication mode is enabled.'));
+    $form_state->setValue(['user_accounts', 'auto_register'], TRUE);
+  }
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function az_cas_form_alter(&$form, FormStateInterface $form_state, $form_id) {
```

Comment:
Can this use [hook_form_BASE_FORM_ID_alter()](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Form%21form.api.php/function/hook_form_BASE_FORM_ID_alter/11.x)?